src/oop/special_class_method/hard/q10/MainTest.java

```diff
@@ -0,0 +1,190 @@
+package oop.special_class_method.hard.q10;
+
+import global.ObjectTest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class MainTest {
+    // Parsons with Distractors
+    private final String className = "Award";
+    private final String firstFieldName = "awardRecipient";
+    private final String secondFieldName = "awardCategory";
+    private final String setSecondMethodName = "setAwardCategory";
+    private final String getSecondMethodName = "getAwardCategory";
+    private final String getFirstMethodName = "getAwardRecipient";
+    private final String setFirstMethodName = "setAwardRecipient";
+    private ObjectTest testClass;
+
+    @BeforeEach
+    public void setUp() {
+        String classString = "oop.special_class_method.hard.q10." + className;
+        testClass = new ObjectTest(classString);
+    }
+
+    @Test
+    public void awardClassHasCorrectAttributes() {
+        assertTrue(testClass.hasField(firstFieldName, String.class),
+                "Your " + className + " class is missing the " + firstFieldName + " attributes.");
+        assertTrue(testClass.hasModifier(firstFieldName, "private"),
+                "Your " + firstFieldName + " does not have the correct visibility modifier.");
+        assertTrue(testClass.hasField(secondFieldName, String.class),
+                "Your " + className + " class is missing the " + secondFieldName + " attributes.");
+        assertTrue(testClass.hasModifier(secondFieldName, "private"),
+                "Your " + secondFieldName + " does not have the correct visibility modifier.");
+    }
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of("First", "Second"),
+                Arguments.of("Science", "Anna"),
+                Arguments.of("Jack", "History"),
+                Arguments.of("Finn", "Fiction")
+
+        );
+    }
+
```

Comment:
There is no test checking if the constructor exists, only that it initializes the attributes correctly and then has the correct modifier